src/ert/callbacks.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Tuple
+
+import xtgeo
+
+from ert._c_wrappers.enkf.enkf_main import field_transform
+from ert._c_wrappers.enkf.enkf_state import _internalize_results
+from ert._c_wrappers.enkf.enums import ErtImplType, RealizationStateEnum
+from ert._c_wrappers.enkf.model_callbacks import LoadStatus
+
+if TYPE_CHECKING:
+    from ert._c_wrappers.enkf import EnsembleConfig, RunArg
+
+logger = logging.getLogger(__name__)
+
+
+def forward_model_ok(
+    run_arg: "RunArg",
+    ens_conf: "EnsembleConfig",
+) -> Tuple[LoadStatus, str]:
+    try:
+        result = (LoadStatus.LOAD_SUCCESSFUL, "")
+
+        if ens_conf.have_forward_init():
+            forward_init_config_nodes = ens_conf.check_forward_init_nodes()
+            error_msg = ""
+            for config_node in forward_init_config_nodes:
+                if not config_node.getUseForwardInit():
+                    continue
+                if config_node.getImplementationType() == ErtImplType.SURFACE:
+                    run_path = Path(run_arg.runpath)
```

Comment:
Seems like `config_node.getImplementationType()` is never equal to `ErtImplType.SURFACE` in the tests.